sources/api/migration/migrator/src/test.rs

```diff
@@ -0,0 +1,275 @@
+        let datastore = Self::create_datastore_links(&tmp, &from_version);
+        TestDatastore { tmp, datastore }
+    }
+
+    /// Migrator relies on the datastore symlink structure to determine the 'from' version.
+    /// This function sets up the directory and symlinks to mock the datastore for migrator.
+    fn create_datastore_links(tmp: &TempDir, from_version: &Version) -> PathBuf {
+        let datastore = tmp.path().join(format!(
+            "v{}.{}.{}_xyz",
+            from_version.major, from_version.minor, from_version.patch
+        ));
+        let datastore_version = tmp.path().join(format!(
+            "v{}.{}.{}",
+            from_version.major, from_version.minor, from_version.patch
+        ));
+        let datastore_minor = tmp
+            .path()
+            .join(format!("v{}.{}", from_version.major, from_version.minor));
+        let datastore_major = tmp.path().join(format!("v{}", from_version.major));
+        let datastore_current = tmp.path().join("current");
+        fs::create_dir_all(&datastore).unwrap();
+        std::os::unix::fs::symlink(&datastore, &datastore_version).unwrap();
+        std::os::unix::fs::symlink(&datastore_version, &datastore_minor).unwrap();
+        std::os::unix::fs::symlink(&datastore_minor, &datastore_major).unwrap();
+        std::os::unix::fs::symlink(&datastore_major, &datastore_current).unwrap();
+        datastore
+    }
+}
+
+/// Represents a TUF repository, which is held in a tempdir.
+struct TestRepo {
+    /// This field preserves the lifetime of the TempDir even though we never read it. When
+    /// `TestRepo` goes out of scope, `TempDir` will remove the temporary directory.
+    _tuf_dir: TempDir,
+    metadata_path: PathBuf,
+    targets_path: PathBuf,
+}
+
+/// LZ4 compresses `source` bytes to a new file at `destination`.
+fn compress(source: &[u8], destination: &Path) {
+    let output_file = File::create(destination).unwrap();
+    let mut encoder = lz4::EncoderBuilder::new()
+        .level(4)
+        .build(output_file)
+        .unwrap();
+    encoder.write_all(source).unwrap();
+    let (_output, result) = encoder.finish();
+    result.unwrap()
+}
+
+/// Creates a test repository with a couple of versions defined in the manifest and a couple of
+/// migrations. See the test description for for more info.
+fn create_test_repo() -> TestRepo {
+    // This is where the signed TUF repo will exist when we are done. It is the
+    // root directory of the `TestRepo` we will return when we are done.
+    let test_repo_dir = TempDir::new().unwrap();
+    let metadata_path = test_repo_dir.path().join("metadata");
+    let targets_path = test_repo_dir.path().join("targets");
+
+    // This is where we will stage the TUF repository targets prior to signing them. It happens
+    // to be the same dir as the root of the tuf_outdir because RepositoryEditor signing uses
+    // symlinks, so we need the tuf_indir and tuf_outdir/targets to both stick around for the
```

Comment:
@zmrow and I are preparing a PR to tough that would add a `copy_targets` call to SignedRepo (like `link_targets`) so this concern might go away.

Reply:
https://github.com/awslabs/tough/pull/158

Reply:
^ has been merged and is available in tough 0.7.0 - does that help you simplify this bit?

Reply:
I don't see it as any more or less complicated. Also, I think it best not to expand the scope of the PR further (to include updating `tough`), so instead I updated the comment to sound less like I had no choice...

[9fcfb47bc](https://github.com/bottlerocket-os/bottlerocket/compare/dd79ba26e48c76ae9d7034ece0753e52b1d9e4db..9fcfb47bc79762c1cd63f86289bd81fef5fb3a9c)

Reply:
I think it's less complicated because you wouldn't need to keep the tempdir around, right?

Reply:
The `TempDir` would still be needed. The difference would be two TempDirs instead of one. Same number of lines of code.